client/js/lounge.js

```diff
@@ -916,29 +927,42 @@ $(function() {
 		}
 	}());
 
-	function completeNicks(word) {
-		const users = chat.find(".active .users");
+	function fuzzyGrep(term, array) {
+		const results = fuzzy.filter(
+			term,
+			array,
+			{
+				pre: "<b>",
+				post: "</b>"
+			}
+		);
+		return results.map((el) => [el.string, el.original]);
+	}
 
-		// Lobbies and private chats do not have an user list
-		if (!users.length) {
-			return [];
-		}
+	function completeNicks(isFuzzy) {
+		return function(word) {
+			const users = chat.find(".active .users");
 
-		const words = users.data("nicks");
+			// Lobbies and private chats do not have an user list
+			if (!users.length) {
+				return [];
+			}
 
-		return $.grep(
-			words,
-			(w) => !w.toLowerCase().indexOf(word.toLowerCase())
-		);
+			const words = users.data("nicks");
+			if (isFuzzy) {
+				return fuzzyGrep(word, words);
+			}
+			return $.grep(
+				words,
+				(w) => !w.toLowerCase().indexOf(word.toLowerCase())
```

Comment:
I just noticed that `word` is lowercased every time in the callback, move `word = word.toLowerCase()` before the `$.grep` call for a performance boost.